### PR TITLE
Remove `mathconstants.h` and use the correct way to get `M_PI` in MSVC.

### PR DESCRIPTION
**Pull request by red-001**
See https://msdn.microsoft.com/en-us/library/4hwaceh6(v=vs.120).aspx

### CODE REVIEW

**Comment by nerzhul** [comment]
@SmallJoker can you test it ?

**Comment by sfan5** [comment]
Shouldn't you be replacing the includes by `#include <math.h>`?

Also the `add_definition()` is placed into the wrong section, put it in the place where Windows compiling f lags are set.

**Comment by red-001** [comment]
`#include <math.h>` doesn't seem to be need as everything builds fine without it.

**Comment by SmallJoker** [comment]
Can confirm. It works.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -284,6 +284,8 @@ if(WIN32)
 		set(PLATFORM_LIBS dbghelp.lib ${PLATFORM_LIBS})
 		# Surpress some useless warnings
 		add_definitions ( /D "_CRT_SECURE_NO_DEPRECATE" /W1 )
+		# Get M_PI to work
+		add_definitions(-D_USE_MATH_DEFINES)
```
Comment (sfan5) [inline src/CMakeLists.txt]: shouldn't this be `add_definitions( /D "_USE_MATH_DEFINES" )` since it's for MSVC?

Comment (SmallJoker) [inline src/CMakeLists.txt]: Both work the same way. The one with slash is MSVC-specific, which actually doesn't matter in this case.

**Comment by red-001** [comment]
Done.

**Pull request merged**